### ISSUE TITLE
Rationale for Undefined Keys in Search Mode

### ISSUE THREAD

**Issue by jkroes** [vim-ctrlspace/vim-ctrlspace#263]
**Is your feature request related to a problem? Please describe.**
Not a problem per se. Every fuzzy search plugin in Vim is different, but certain features are shared by multiple plugins. In particular, many fuzzy searchers can include punctuation characters in a search query, either as literal characters or special characters for regex-based queries. I was surprised that this plugin generates an error message " Key <key> doesn't work in this view." I managed to make some test queries search for literal period characters with

```vim
let g:CtrlSpaceKeys = {'Search': {'.': 'ctrlspace#keys#search#AddLetter'}}
```

My questions:
1. Is there any reason I shouldn't do this? There must be a reason this isn't enabled by default, right? 
2. Is it possible to do regex queries using a period as a placeholder for any character, or maybe even globbed queries? 
3. Related to my second question, regex-based queries would make finding an exact filename (at any path) possible. E.g., searching for vim.vim$ to find a file by that name rather than say, **/vim/**/python.vim. 

**Describe the solution you'd like**
An explanation of the default keymappings, advice on altering it to match punctuation literally, and whether regex-based fuzzy searches can be enabled. 

**Describe alternatives you've considered**
There really isn't an alternative if you want integration with this package. Of course, the behavior described above could be added by installilng another fuzzy plugin to open files within the current tab, then using ctrlspace to manage the tab. 

**Version(s) (please complete the following information):**
 - OS: 
Ubuntu 18.04 (technically the ubuntu:18.04 docker image)
 - :version
NVIM v0.4.3
Build type: RelWithDebInfo
LuaJIT 2.0.5
Compilation: /usr/bin/gcc-5 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DMIN_LOG_LEVEL=3 -Og -g -Wall -Wextra -pedan
tic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wvla -fstack-prot
ector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM
_UNIBI_HAS_VAR_FROM -I/home/travis/build/neovim/bot-ci/build/neovim/build/config -I/home/travis/build/neovim/bot-ci/bu
ild/neovim/src -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/usr/include -I/home/travis/build/n
eovim/bot-ci/build/neovim/build/src/nvim/auto -I/home/travis/build/neovim/bot-ci/build/neovim/build/include
Compiled by travis@travis-job-9a287301-34f2-4b86-92e3-462594f51f36

Features: +acl +iconv +tui

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/home/travis/build/neovim/bot-ci/build/neovim/build/nvim.AppDir/usr/share/nvim"

**Comment by Konfekt** [comment]
Good point. From what I can see at

```vim
" autoload/ctrlspace/keys/search.vim (line 12)
call ctrlspace#keys#AddMapping("ctrlspace#keys#search#AddLetter",           "Search", ["lowercase", "uppercase", "numbers"])
```

it seems an omission, and perhaps we could add other printable characters to

```vim
" autoload/ctrlspace/keys.vim (line 52)
function! s:initKeyNames()
```

that allow a mapping in Vim that is set up

```vim
" autoload/ctrlspace/keys.vim (line 118)
function! ctrlspace#keys#AddMapping(funcName, mapName, keys)
```

Unfortunately, non-ASCII characters are therefore not readily mapped. But at least, say, all punctuation characters should be added.

**Comment by jyscao** [comment]
I fully agree that the default keys should include some sensible punctuation chars.

**Comment by Konfekt** [comment]
@jkroes Did the merged PR sufficiently address this issue?
